octodns_edgecenter/__init__.py

```diff
@@ -443,30 +526,63 @@ def _should_ignore(self, record):
                     want_type,
                 )
                 return True
-        if want_filters == 2:
-            return False
-        limits = [filters[i].get("limit", 1) for i in [1, 2]]
-        if limits[0] != limits[1]:
+
+        limits = {fls.get("limit") for fls in filters if "limit" in fls}
+        if limits and len(limits) != 1:
             self.log.info(
-                "ignore %s, filters.1.limit (%d) != filters.2.limit (%d)",
-                name,
-                limits[0],
-                limits[1],
+                "ignore %s, filters have different limit values", name
             )
             return True
+
         return False
 
+    @staticmethod
+    def _params_for_failover(record: Record) -> Dict:
+        config = dict()
+        additional_data = record.octodns.get("edgecenter", {}).get(
+            "failover", {}
+        )
+
+        # TODO: remove ignored coverage check after add support ICMP in octodns
+        if record.healthcheck_protocol != "ICMP":  # pragma: no branch
```

Comment:
1.9.0 is out, should show up for CI jobs soon, though i've seen things take a few hours in github actions as of late https://github.com/octodns/octodns/pull/1184

```suggestion
        if record.healthcheck_protocol != "ICMP":
```

Reply:
I've updated the requirements with a script to use 1.9.0. Since the version supports UDP and ICMP protocols, I added tests for these cases.